cmake/Config.cmake

```diff
@@ -169,6 +169,7 @@ cxx_add_compile_options(GNU
     --std=c++11
     -Wall
     -Werror
+    -Wno-deprecated-declarations
```

Comment:
What are we using that's deprecated?

Reply:
Boost uses deprecated `std::auto_ptr`. Will add a comment here.